src/HtmlGenerator/Program.cs

```diff
@@ -26,18 +29,38 @@ static void Main(string[] args)
                     continue;
                 }
 
-                try
+                if (arg.StartsWith("/in:"))
                 {
-                    var project = Path.GetFullPath(arg);
-                    if (File.Exists(project))
+                    string inputPath = arg.Substring("/in:".Length).StripQuotes();
+                    try
                     {
-                        if (project.EndsWith(".sln", StringComparison.OrdinalIgnoreCase) ||
-                            project.EndsWith(".csproj", StringComparison.OrdinalIgnoreCase) ||
-                            project.EndsWith(".vbproj", StringComparison.OrdinalIgnoreCase))
+                        if (!File.Exists(inputPath))
                         {
-                            projects.Add(project.StripQuotes());
+                            continue;
+                        }
+                        using (StreamReader reader = new StreamReader(inputPath))

```

Comment:
I think File.ReadAllText conveys the intent better, is easier to read and understand and results in slightly shorter code.

-----Original Message-----
From: "David Rose" <notifications@github.com>
Sent: ‎10/‎3/‎2015 12:03 PM
To: "KirillOsenkov/SourceBrowser" <SourceBrowser@noreply.github.com>
Cc: "Kirill Osenkov" <github@osenkov.com>
Subject: Re: [SourceBrowser] Added ability to specify an input file (#12)

In src/HtmlGenerator/Program.cs:
>                          {
> -                            projects.Add(project.StripQuotes());
> +                            continue;
> +                        }
> +                        using (StreamReader reader = new StreamReader(inputPath))
I've made the change, although would you be able top explain why this would be preferred?
—
Reply to this email directly or view it on GitHub.